### DISCUSSION TITLE
big error in tensile stress computation using dp

### DISCUSSION THREAD

**Discussion by duanke1359** [deepmodeling/deepmd-kit#1872] [Q&A] [answered]
Hi, I encoutered two problems when I perform simulations using self-trained dp. My source simulation file is attached as Problem.zip. I would be very grateful if you can give some help.
Problem 1
I try to validate my trained dp by performing axial tensile simulation on a (10,10) CNT. The simulation results by classical potential AIREBO was used for comparison. As shown in Figure 1, the configuration of CNT holds well by our dp. However, when I extract the results of strain energy and axial tensile stress, something strange can be observed. In left one figure 2, we can see that the strain energy versus strain curves show well consistency between dp and airebo. But in right one of figure 2, the tensile stress of airebo is right, which corresponds to its strain energy variation, while for dp, there is almost of no tensile stress. For the two simulations, all lammps settings are kept the same except the usage of potential. The version of dp-kit is 2.0.3.
![image](https://user-images.githubusercontent.com/34574385/185890149-5a13e964-7d85-41f3-b216-9e55bc2720c6.png)
Problem 2:
Lammps simulations runs well for a small model of 10675 atoms, while report errors and terminated when the simulation model becomes bigger of 33856 atoms. Why this problem happened, then how can I address such issue. The version of dp-kit is v2.0.3.
![image](https://user-images.githubusercontent.com/34574385/185890233-fee0bb2b-437d-4aba-8856-3256ff2caaba.png)
[Problem.zip](https://github.com/deepmodeling/deepmd-kit/files/9393105/Problem.zip)

**Comment by njzjz** [comment, marked as answer]
How did you calculate tensile stress?

Problem 2 was discussed in #1062. It is a bug in CUDA Toolkit 11.1~11.3.

**Comment by AnguseZhang** [comment]
Has dicussed with @duanke1359 , the first problem is related to the interface with LAMMPS when calculating atomic virials. 
https://docs.deepmodeling.com/projects/deepmd/en/master/third-party/lammps-command.html#use-of-the-centroid-stress-atom-to-get-the-full-3x3-atomic-virial

**Reply by njzjz** [reply]
Is it a bug of LAMMPS or our plugin?

**Reply by AnguseZhang** [reply]
In a common potential, like EAM, the command to get atomic virial is `compute stress/atom`. But in DeePMD-kit pair_style, it should be `compute centroid/stress/atom`.  I'm not clear about the purpose to design such interface, @amcadmus do you have any suggestions?

**Reply by wanghan-iapcm** [reply]
the standard lammps virial command does not support asymmetric atomic virial tensors, which are required for many-body potentials like DP.
